docs/aws/audit/sgaudit/rules/ec2_instances_exist_with_ports_open_for_internal_remediation.mdx

```diff
@@ -1,6 +1,242 @@
+2. In the left navigation pane, under the "Network & Security" section, click on "Security Groups".
+3. In the Security Groups page, you will see a list of all security groups associated with your AWS account. Click on the security group that you want to inspect.
+4. In the details pane at the bottom, click on the "Inbound" tab. This will show you all the inbound rules associated with the selected security group. If there are any rules that allow traffic from all IP addresses (0.0.0.0/0) on any port, then the port is open for internal traffic.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. **Install and Configure AWS CLI**: Before you can start using AWS CLI, you need to install it on your local machine. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. **List All Security Groups**: Use the `describe-security-groups` command to list all the security groups in your AWS account. The command is as follows:
+
+    ```
+    aws ec2 describe-security-groups --query "SecurityGroups[*].[GroupId,GroupName]" --output text
+    ```
+
+    This command will return a list of all security groups along with their Group ID and Group Name.
+
+3. **Check Security Group Rules**: For each security group, you need to check the inbound and outbound rules. You can do this using the `describe-security-groups` command again, but this time you need to specify the Group ID. The command is as follows:
+
+    ```
+    aws ec2 describe-security-groups --group-ids <Group-ID>
+    ```
+
+    Replace `<Group-ID>` with the ID of the security group you want to check. This command will return a list of all inbound and outbound rules for the specified security group.
+
+4. **Identify Open Ports**: From the output of the previous command, you need to identify any rules that allow traffic from 0.0.0.0/0 (all IP addresses). These are the open ports. Specifically, look for the `IpRanges` field in the output. If it contains an entry with `CidrIp` set to 0.0.0.0/0, then that port is open to all IP addresses.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. **Import necessary libraries**: The first step is to import the necessary libraries in your Python script. You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, and others.
+
+```python
+import boto3
+```
+
+2. **Create a session**: The next step is to create a session using your AWS credentials. You can do this by using the Session function in boto3. Replace 'your_access_key', 'your_secret_key', and 'your_region' with your actual AWS credentials.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='your_access_key',
+    aws_secret_access_key='your_secret_key',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>